src/main/java/fi/fmi/avi/converter/tac/lexer/impl/token/AdvisoryRemarkStart.java

```diff
@@ -0,0 +1,35 @@
+package fi.fmi.avi.converter.tac.lexer.impl.token;
+
+import static fi.fmi.avi.converter.tac.lexer.LexemeIdentity.REMARKS_START;
+
+import java.util.Optional;
+
+import fi.fmi.avi.converter.ConversionHints;
+import fi.fmi.avi.converter.tac.lexer.Lexeme;
+import fi.fmi.avi.converter.tac.lexer.SerializingException;
+import fi.fmi.avi.converter.tac.lexer.impl.FactoryBasedReconstructor;
+import fi.fmi.avi.converter.tac.lexer.impl.PrioritizedLexemeVisitor;
+import fi.fmi.avi.converter.tac.lexer.impl.ReconstructorContext;
+import fi.fmi.avi.model.AviationWeatherMessageOrCollection;
+
+public class AdvisoryRemarkStart extends PrioritizedLexemeVisitor {
+    public AdvisoryRemarkStart(final PrioritizedLexemeVisitor.OccurrenceFrequency prio) {
+        super(prio);
+    }
+
+    @Override
+    public void visit(final Lexeme token, final ConversionHints hints) {
+        if ("RMK:".equals(token.getTACToken())) {
+            token.identify(REMARKS_START);
+        }
+    }
+
+    public static class Reconstructor extends FactoryBasedReconstructor {
+        @Override
+        public <T extends AviationWeatherMessageOrCollection> Optional<Lexeme> getAsLexeme(final T msg, final Class<T> clz, final ReconstructorContext<T> ctx)
+                throws SerializingException {
+            return Optional.of(this.createLexeme(new StringBuilder("RMK:").toString(), REMARKS_START));
```

Comment:
I would clean up the `StringBuilder` stuff away.

Reply:
The string builder has been removed. The return value is now the plain string